src/weaponstats.cc

```diff
@@ -0,0 +1,171 @@
+#include "weaponstats.hh"
+#include "functions.hh"
+#include <cstdlib>
+#include <algorithm>
+#include "injector/injector.hpp"
+#include "logger.hh"
+#include "base.hh"
+#include <iostream>
+#include "config.hh"
+
+int                    statsStart                       = 0xC8AAB8;
+int                    itemSize                         = 0x70;
+WeaponStatsRandomizer *WeaponStatsRandomizer::mInstance = nullptr;
+int                    hexFlags[]
+    = {0x3033, 0x3833, 0x7001, 0x7011,  0x7031,  0x2001, 0x2011, 0x2031, 0x3013,
+       0x3033, 0xA008, 0xA014, 0x48214, 0x30238, 0x238,  0x0038, 0x0431};
```

Comment:
I'm not sure what this is supposed to mean, could you elaborate a bit on what it is?

Reply:
I created an array with all hex values of weapon properties that are used in weapons.dat file so it would be easier to randomize them. Every of them are defining a things like ability to sprint with certain weapon, ability to crouch, aiming mode and if weapon have akimbo. So for example you can have AK-47 with properties of pistol akimbo.

Reply:
Ah alright, makes sense. Thanks.